test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetPackCommandTest.cs

```diff
@@ -2182,6 +2182,181 @@ public void PackCommand_ReferencedProjectWithCustomTokensInNuspec()
+                };
+
+                // Act
+                var exception = Record.Exception(() =>
+                {
+                    var runner = new PackCommandRunner(args, ProjectFactory.ProjectCreator);
+                    _ = runner.RunPackageBuild();
+                });
+
+                // Assert
+                Assert.Null(exception);
+            }
+        }
+
+        /// <summary>
+        /// Creates a simple project that defines attributes commonly found in AssemblyInfo.cs
+        /// </summary>
+        /// <remarks>
+        /// The project is created under directory baseDirectory\projectName.
+        /// The project contains just one file called file1.cs.
+        /// </remarks>
+        /// <param name="baseDirectory">The base directory.</param>
+        /// <param name="projectName">The name of the project.</param>
+        /// <param name="referencedProject">The list of projects referenced by this project. Can be null.</param>
+        /// <param name="targetFrameworkVersion">The target framework version of the project.</param>
+        /// <param name="version">The text for the AssemblyVersion attribute.</param>
+        /// <param name="fileVersion">The text for the AssemblyFileVersion attribute.</param>
+        /// <param name="informationalVersion">The text for the AssemblyInformationalVersion attribute.</param>
+        private void CreateTestProjectWithAssemblyInfoAndNuspec(
+            string baseDirectory,
+            string projectName,
+            string version,
+            string fileVersion,
+            string informationalVersion)
+        {
+            var projectContent = @"<Project ToolsVersion='4.0' DefaultTargets='Build'
+    xmlns='http://schemas.microsoft.com/developer/msbuild/2003'>
+  <PropertyGroup>
+    <OutputType>Library</OutputType>
+    <OutputPath>out</OutputPath>
+    <TargetFrameworkVersion>v4.0</TargetFrameworkVersion>
+  </PropertyGroup>
+  <ItemGroup>
+    <Compile Include='file1.cs' />
+  </ItemGroup>
+  <Import Project='$(MSBuildToolsPath)\Microsoft.CSharp.targets' />
+</Project>";
+
+            var projectDirectory = Path.Combine(baseDirectory, projectName);
+            _ = Directory.CreateDirectory(projectDirectory);
+            Util.CreateFile(projectDirectory, projectName + ".csproj", projectContent);
+
+            var csharpContent = @"using System;
+using System.Reflection;
+using System.Runtime.InteropServices;
+
+[assembly: AssemblyTitle(""The Title"")]
+[assembly: AssemblyDescription(""The Description"")]
+[assembly: AssemblyConfiguration("""")]
+[assembly: AssemblyCompany(""The Company"")]
+[assembly: AssemblyProduct(""The Product"")]
+[assembly: AssemblyCopyright(""The Copyright"")]
+[assembly: AssemblyTrademark("""")]
+[assembly: AssemblyCulture("""")]
+
+[assembly: AssemblyVersion(""[[version]]"")]
+[assembly: AssemblyFileVersion(""[[file-version]]"")]
+[assembly: AssemblyInformationalVersion(""[[informational-version]]"")]
```

Comment:
You may refer to https://github.com/NuGet/NuGet.Client/blob/dev/test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Caching/CachingTestContext.cs#L289-L291,
using variables in the string.